src/EditorFeatures/CSharpTest/NavigateTo/NavigateToSearcherTests.cs

```diff
@@ -80,22 +58,18 @@ public class NavigateToSearcherTests
                     pattern,
                     ImmutableHashSet<string>.Empty,
                     It.IsAny<Document?>(),
-                    It.IsAny<Func<INavigateToSearchResult, Task>>(),
-                    It.IsAny<CancellationToken>())).Callback(
-                    (Project project,
-                     ImmutableArray<Document> priorityDocuments,
-                     string pattern2,
-                     IImmutableSet<string> kinds,
-                     Document? activeDocument,
-                     Func<INavigateToSearchResult, Task> onResultFound2,
-                     CancellationToken cancellationToken) =>
-                    {
-                        if (result != null)
-                            onResultFound2(result);
-                    }).Returns(Task.CompletedTask);
+                    It.IsAny<CancellationToken>())).Returns(GetEnumerable(result));
             }
         }
 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
This seems odd: can we delete the async keyword here, or is this a compiler bug?

Reply:
no.  unfortunately, you must use `async` to use `yield` to create an IAsyncEnumerbale (not sure why tbh).  So you have to add that modifier, then you have to suppress the warning.  @jcouv ?